tools/mkclass/classcompiler.cpp

```diff
@@ -385,7 +385,10 @@ void ClassCompiler::HandleClass(const Klass& klass, const ClassDebugInfo&)
 	for (const std::string& dep : klass.LoadDependencies)
 		m_Impl << "\t\t" << "GetByName(\"" << dep << "\").get()," << std::endl;
 
-	m_Impl << "\t" << "});" << std::endl;
+	m_Impl << "\t});" << std::endl << std::endl;
+
+	m_Impl << "\tstatic const auto v (VERIFY(deps.find(nullptr) == deps.end()));" << std::endl;
+	m_Impl << "\t(void)v;" << std::endl << std::endl;
```

Comment:
I wouldn't mind doing this check elsewhere, as its done in #10148, but if you want to do it here, then please put this inside the loop above and perform the sanity check for each dependency type and then throw a proper exception, instead of such cryptic and mysterious death that nobody understands.
```bash
/Users/yhabteab/Workspace/icinga2/build/lib/remote/apilistener-ti.cpp:211: assertion failed: deps.find(nullptr) == deps.end()
[1]    68858 abort      prefix/sbin/icinga2 daemon
```